spec/declaration.dd

```diff
@@ -502,6 +503,75 @@ a = 2;         // sets `S.i` to `2`
 b = 4;         // sets `S.j` to `4`
 -----------
 
+$(H2 $(LNAME2 AliasAssign, Alias Assign))
+
+$(GRAMMAR
+$(GNAME AliasAssign):
+    $(I Identifier) $(D =) $(GLINK Type)
+)
+
+        $(P An $(GLINK AliasDeclaration) can have a new value assigned to it with an
+        Alias Assign:)
+
+---
+template Gorgon(T)
+{
+    alias A = long;
+    A = T;
+    alias Gorgon = A;
+}
+pragma(msg, Gorgon!int); // prints int
+---
+
+$(UL
+$(LI The $(I AliasAssign) and its corresponding $(I AliasDeclaration) must both be
+declared in the same $(GLINK2 template, TemplateDeclaration).)
+
+$(LI The corresponding $(I AliasDeclaration) must appear lexically before the
+$(I AliasAssign).)
+
+$(LI The corresponding $(I AliasDeclaration) may not refer to overloaded symbols.)
+
+$(LI The value of an $(I AliasDeclaration) or $(I AliasAssign) may not be used prior
+to an $(I AliasAssign) other than in the right hand side of an $(I AliasAssign) for it.)
```

Comment:
This needs rewording. It's confusing. Are the three `AliasAssigns` distinct, or are there two?  And what is the "it" referring to? I'm not fully grokking it, so I have no suggestions.

Reply:
done